### ISSUE TITLE
Gutentag creates 'empty string' tags

### ISSUE THREAD

**Issue by cernyjakub** [pat/gutentag#51]
Hello, thanks for a nice gem!

Just noticed that Gutentag happily creates a tag with `''` value.

I suggest adding a default validation for minimum length of 1. I cannot imagine using a tag with blank value.

These values are sometimes submitted from forms generated by Formtastic and similar gems - they add hidden field to make sure data is sent when all tags are cleared.

What do you think?

**Comment by pboling** [comment]
It would result in incorrectly linked all kinds of unintended things that have "no" tags theoretically, via a zombie blank tag.

**Comment by cernyjakub** [comment]
OK, you suggest to keep the behaviour. From my perspective it is a little awkward to link objects into groups by zombie tags... but no problem ;)  I have created a before_action filter in my app, which strips blank values from tag_names array submitted from forms.

Anyway, thanks for reply!
Jakub

**Comment by pat** [comment]
@cernyjakub I'm not against such a patch… I'd like to understand a bit better about how a form passes through the data that results in empty tags. Does it occur as part of translating a string of comma-separated values into an array of string names?

@pboling I'm not quite following your point - the suggestion from Jakub was to get rid of the zombie blank tags, which makes sense to me. Is there something I'm missing?

**Comment by cernyjakub** [comment]
Hi @pat!

In my app, I don't submit comma separated tags in single string. I use _formtastic gem_ powered form like this (HAML fragment from the form):

`= f.input :tag_names, :collection => ['tag1', 'tag2', 'tag3'], :as => :check_boxes`

This shows user a series of check boxes to choose from (I only allow limited set of tags to add here). However, formtastic adds one more hidden field called `mymodel[tag_names][]` with empty value, which is submitted. I think the reason is to sent info about de-selected checkboxes to the server (if you clear all of them, the value would not be sent at all and server model not updated). If I am right this is standard behaviour of Rails form collection helpers, as well as form gems like `formtastic` and `simple_form`, etc.

That's the whole story. Server receives array with blank first element:  
`"tag_names"=>["", "user_detail_interview"]`

.. and Gutentag's setter method `tag_names=` assigns empty string into the tag list - creating tag with `name = ''`.

For example `acts_as_taggable_on` gem eliminates blank tags, see (from my other app):
```ruby
a.tag_list = ["", "test"]
=> ["", "test"]
>> a.tag_list
=> ["test"]
```
So I think it would be logical, to add the same behaviour into Gutentag too. Sorry for long post ;)

regards!
Jakjb

**Comment by pat** [comment]
No need to apologise, that context is super helpful and makes sense. I think you're right that blank tags have no value. I'll see what I can do about getting the gem to ignore those values.

Though @pboling if you think that new behaviour wouldn't be correct, I'm keen to hear more as to why! :)

**Comment by pat** [comment]
@cernyjakub Just pushed 9c6060095aec4bda0cb0a1c9cf7ec6cccc52070c - can you confirm that works well for your situation? :)

**Comment by cernyjakub** [comment]
Wonderful! Worked like a charm for me. Thanks!

**Comment by pat** [comment]
Good to know :) I've set a reminder for me to release a new version of Gutentag in about a week's time (depending on any other patches/issues that crop up before then).

**Comment by pboling** [comment]
No, sorry I was agreeing, and it all came out wrong.  I was trying to say that a blank "" tag would be a zombie and get linked to many things that were unintended due to peculiarities of form processing.

👍 Love the change!  Will try it out soon.

**Comment by pat** [comment]
Appreciate the clarification @pboling - thanks :) and good to know we're all on the same page!

**Comment by pat** [comment]
I've just published v2.3.0 which includes this fix :)
